### ISSUE TITLE
Extract logic for constructing from JSON or provide sync overload

### ISSUE THREAD

**Issue by danielwertheim** [AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet#303]
Need a syncronous flow as I'm in an environment where I can not be async all the way.

For this reason it would be great if the actual logic here for constructing and populating the Config instance could be extracted to a factory or builder or something: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/KDev/src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfigurationRetriever.cs#L80

That way, we could extract the JSON in whatever way we want and construct a config object.

Or provide a truly syncronous `IDocumentRetriever` definition.

**Comment by brentschmaltz** [comment]
@danielwertheim you can construct a configuration object from a JSON string. You can then use the StaticConfigurationManager accessing the Config.

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/KDev/src/Microsoft.IdentityModel.Protocols/Configuration/StaticConfigurationManager.cs#L48

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/KDev/src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfiguration.cs#L75

Would this work?

**Comment by danielwertheim** [comment]
I'm aware of that construct. I'm using it. The thing is that since I'm in a pure syncronous flow, and refuse to try and get sync over async to work, I had to write my own syncronhous fetcher. Then neither of the `ConfigurationManager` implementations then helps. But since the code is open source (awesome), I've been able to see how you extract the `X5092Certificate` so I'm able to work without dependency on NewtonSoft etc.
